libs/orders/src/lib/components/order-list/order-edit-dialog.tsx

```diff
@@ -88,15 +88,24 @@ export const OrderEditDialog = ({
         <form onSubmit={handleSubmit(onSubmit)} data-testid="edit-order">
           <FormGroup label={t('Entry price')} labelFor="entryPrice">
             <Input
-              {...register('entryPrice', { required: t('Required') })}
+              type="number"
+              step={step}
+              {...register('entryPrice', {
+                required: t('You need to provide a price'),
+                min: 0,
+              })}
               id="entryPrice"
-              type="text"
             />
             {errors.entryPrice?.message && (
               <InputError intent="danger" className="mt-4">
                 {errors.entryPrice.message}
               </InputError>
             )}
+            {errors.entryPrice?.type === 'min' && (
```

Comment:
chore (blocking): You should be able to just render the message once here provided you have set up the validation using the register function. Check out the react-hook-form docs..

You should be able to do something like
```
register('entryPrice', {
  validation: {
    min: (value) => {
      if (value is valid) return true
      return 'The price cannot be negative'
    } 
  }
})
```

Reply:
updated